clang/lib/CodeGen/CGCall.cpp

```diff
@@ -4681,6 +4681,11 @@ void CallArg::copyInto(CodeGenFunction &CGF, Address Addr) const {
   IsUsed = true;
 }
 
+void CodeGenFunction::EmitWritebacks(CodeGenFunction &CGF,
+                                     const CallArgList &args) {
+  emitWritebacks(CGF, args);
```

Comment:
If the C++ ABI's arguments are destroyed left-to-right the writebacks need to be reversed before this is called. Something like:
 
 ```suggestion
  if (getTarget().getCXXABI().areArgsDestroyedLeftToRightInCallee())
    Args.reverseWritebacks();
  emitWritebacks(CGF, args);
```